src/transformers/trainer.py

```diff
@@ -220,6 +225,12 @@ def get_optimizers(
         )
         return optimizer, scheduler
 
+    def _setup_wandb(self):
+        # Start a wandb run and log config parameters
+        wandb.init(name=self.args.logging_dir, config=vars(self.args))
+        # keep track of model topology and gradients
+        # wandb.watch(self.model)
```

Comment:
This line caused a crash for me later on so I commented it out @borisdayma 

In any case, one can override this method to customize the wandb initialization if needed.

Reply:
What it does is logging gradients and model topology as in [this run](https://app.wandb.ai/borisd13/transformers-examples/runs/3iuvnwam?workspace=user-borisd13). It's not essential but can be interesting.
Do you have an example of where it failed?
Note: this needs to be called after `wandb.init`.

Reply:
Tried to reproduce today (and open an issue on https://github.com/wandb/client) but for some reason, couldn't – so I'll uncomment this.